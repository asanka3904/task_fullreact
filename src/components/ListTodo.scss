.todo-list {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;

  .todo-list-header {
    display: flex;
    background: white;
    background: linear-gradient(
      to right bottom,
      rgba(255, 255, 255, 0.8),
      rgba(255, 255, 255, 0.3)
    );
    margin: 0;
    padding-left: 20px;
    padding-right: 20px;

    .line {
      display: flex;
      margin: 2px;
      padding: 2px;
    }
  }

  .list {
    width: 100%;
    max-width: 360px;
    background: rgb(79, 125, 129);
    background: linear-gradient(
      to right bottom,
      rgba(255, 255, 255, 0.8),
      rgba(255, 255, 255, 0.3)
    );
    border-radius: 1rem;
  }
}
