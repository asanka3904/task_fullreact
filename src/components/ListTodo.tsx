import React, { useEffect, useState } from "react";

import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import ListItemText from "@material-ui/core/ListItemText";
import Checkbox from "@material-ui/core/Checkbox";
import IconButton from "@material-ui/core/IconButton";
import DeleteOutlinedIcon from "@material-ui/icons/DeleteOutlined";

import ArrowUpwardOutlinedIcon from "@material-ui/icons/ArrowUpwardOutlined";
import ArrowDownwardOutlinedIcon from "@material-ui/icons/ArrowDownwardOutlined";

import { useAppSelector, useAppDispatch } from "../app/hooks";
import {
  removeTodo,
  selectTodo,
  setTodoStatus,
} from "../features/todo/todoSlice";

import { Todo } from "../interface/Todo";

import "./ListTodo.scss";

export const ListTodo = () => {
  const dispatch = useAppDispatch();
  const todolist = useAppSelector(selectTodo);
  console.log(todolist);

  const [task, settask] = useState<Todo[]>([]);

  let arr: Todo[] = [];

  useEffect(() => {
    settask(todolist);
  }, [todolist]);

  //short descending

  const sortdes = (e: any) => {
    e.preventDefault();
    settask([]);
    var compareDate = (t1: Todo, t2: Todo) => {
      var t1Date = new Date(t1.createDate);
      var t2Date = new Date(t2.createDate);
      return t1Date > t2Date ? -1 : 1;
    };

    arr = [...todolist].sort(compareDate);

    settask(arr);
  };

  //sort ascending
  const sortas = (e: any) => {
    e.preventDefault();
    settask([]);
    var compareDate = (t1: Todo, t2: Todo) => {
      var t1Date = new Date(t1.createDate);
      var t2Date = new Date(t2.createDate);
      return t1Date > t2Date ? 1 : -1;
    };

    arr = [...todolist].sort(compareDate);

    settask(arr);
  };

  return (
    <div className="todo-list">
      <div className="todo-list-header">
        <div className="line">
          <IconButton edge="end" aria-label="delete" onClick={sortdes}>
            <ArrowUpwardOutlinedIcon />
          </IconButton>
          <IconButton edge="end" aria-label="delete" onClick={sortas}>
            <ArrowDownwardOutlinedIcon />
          </IconButton>
        </div>
      </div>

      <List className="list">
        {task.map((todo) => {
          const labelId = `checkbox-list-label-${todo.id}`;

          return (
            <ListItem key={todo.id} role={undefined} dense>
              <ListItemIcon>
                <Checkbox
                  edge="start"
                  value={todo.complated}
                  onChange={() => {
                    dispatch(
                      setTodoStatus({ complated: !todo.complated, id: todo.id })
                    );
                  }}
                />
              </ListItemIcon>
              <ListItemText
                id={labelId}
                primary={todo.title}
                style={{
                  textDecoration: todo.complated ? "line-through" : "none",
                  color: todo.complated ? "rgba(184, 172, 172, 0.89)" : "black",
                }}
              />
              <ListItemSecondaryAction>
                <IconButton
                  edge="end"
                  aria-label="delete"
                  onClick={() => {
                    dispatch(removeTodo(todo.id));
                  }}
                >
                  <DeleteOutlinedIcon />
                </IconButton>
              </ListItemSecondaryAction>
            </ListItem>
          );
        })}
      </List>
    </div>
  );
};
