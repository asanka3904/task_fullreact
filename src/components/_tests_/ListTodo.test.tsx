import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ListTodo } from "../ListTodo";
import { Provider } from "react-redux";
import { store } from "../../app/store";

//test the list unit
test("ListTodo", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <ListTodo />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
