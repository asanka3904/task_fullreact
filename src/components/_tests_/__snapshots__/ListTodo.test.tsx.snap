// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTodo 1`] = `
<div
  className="todo-list"
>
  <div
    className="todo-list-header"
  >
    <div
      className="line"
    >
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
            />
          </svg>
        </span>
      </button>
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root list MuiList-padding"
  />
</div>
`;
